homeassistant/components/tplink/__init__.py

```diff
@@ -220,6 +220,7 @@ def __init__(
     ) -> None:
         """Initialize DataUpdateCoordinator to gather data for specific SmartPlug."""
         self.smartplug = smartplug
+        self.has_emeter = self.smartplug.has_emeter
```

Comment:
Does this do I/O?

Reply:
Yes, this does I/O.

Reply:
It has to be a corutine function since the coordinator uses that interface. We need to schedule all I/O on the executor and we don't want any side effects in the init method here.

Reply:
Makes sense, I'm working on it (moving the updates out from `_async_update_data` and using the executor to execute it). I also noticed that the `__init__` is already doing I/O (accessing `smartplug.alias`), but I'm unsure what's the solution for that.

Reply:
> smartplug.alias

Do the call before creating the `SmartPlugDataUpdateCoordinator` in the executor, and pass it in

Reply:
I re-used the data from the update response for this purpose. It's not very useful as it's the name of the device and not the name of the socket, but I suppose that's fine for now. Feel free to give it a test @bdraco :-)